modules/checklist/assets/js/app/components/header.js

```diff
@@ -2,45 +2,35 @@ import { Typography, CloseButton, AppBar, Divider, Toolbar } from '@elementor/ui
 import { __ } from '@wordpress/i18n';
 import Progress from './progress';
 import PropTypes from 'prop-types';
-import { QueryClient, QueryClientProvider, useQuery } from '@elementor/query';
+import { useQuery } from '@elementor/query';
 import * as React from 'react';
+import { toggleChecklistPopup } from '../../utils/functions';
+import { USER_PROGRESS } from '../../utils/consts';
+
+const { FIRST_CLOSED_CHECKLIST_IN_EDITOR } = USER_PROGRESS;
 
 const fetchStatus = async () => {
-	const response = await fetch( `${ elementorCommon.config.urls.rest }elementor/v1/checklist/user-progress`, {
-		method: 'GET',
-		headers: {
-			'Content-Type': 'application/json',
-			'X-WP-Nonce': elementorWebCliConfig.nonce,
-		},
-	} );
-	const data = await response.json();
-	return data.data.first_closed_checklist_in_editor;
-};
+	const response = await $e.data.get( 'checklist/user-progress', {}, { refresh: true } );
 
-const queryClient = new QueryClient();
+	return response?.data?.data?.[ FIRST_CLOSED_CHECKLIST_IN_EDITOR ] || false;
+};
 
-const HeaderContent = ( { steps } ) => {
-	const { error, data: closedForFirstTime } = useQuery( {
+const Header = ( { steps } ) => {
+	const { data: closedForFirstTime } = useQuery( {
 		queryKey: [ 'closedForFirstTime' ],
 		queryFn: fetchStatus,
 	} );
 
-	if ( error ) {
-		return null;
-	}
-
 	const closeChecklist = async () => {
-		if ( closedForFirstTime !== true ) {
-			await fetch( `${ elementorCommon.config.urls.rest }elementor/v1/checklist/user-progress`, {
-				method: 'POST',
-				headers: {
-					'Content-Type': 'application/json',
-					'X-WP-Nonce': elementorWebCliConfig.nonce,
-				},
+		if ( ! closedForFirstTime ) {
+			await $e.data.update( 'checklist/user-progress', {
+				[ FIRST_CLOSED_CHECKLIST_IN_EDITOR ]: true,
```

Comment:
Maybe?
```suggestion
				[ CHECKLIST_CLOSED_IN_THE_EDITOR_FOR_FIRST_TIME ]: true,
```
From this text we can understand what this const means.

Reply:
Sure!